File: Data Structures & Algorithms/validate-parentheses/submission-5.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
        const closeToOpen = {
            ")": "(",
            "]": "[",
            "}": "{",
        }

        const stack = [];
        for (const c of s) {
            if (closeToOpen[c]) {
                if (stack.length > 0 && stack[stack.length - 1] === closeToOpen[c]) {
                    stack.pop();
                } else {
                    return false;
                }
            } else {
                stack.push(c);
            }
        }
        return stack.length === 0;
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-5.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        for tok in tokens:
            if tok == "+":
                stack.append(stack.pop() + stack.pop())
            elif tok == "*":
                stack.append(stack.pop() * stack.pop())
            elif tok == "-":
                val1 = stack.pop()
                val2 = stack.pop()
                stack.append(val2 - val1)
            elif tok == "/":
                val1 = stack.pop()
                val2 = stack.pop()
                stack.append(int(val2 / val1)) # disgusting!
            else:
                stack.append(int(tok))
        return stack[-1]